client/app/services/query.js

```diff
@@ -285,11 +285,7 @@ class Parameters {
   }
 
   toUrlParams() {
-    if (this.get().length === 0) {
-      return '';
-    }
-
-    return qs.toString(Object.assign(...this.get().map(p => p.toUrlParams())));
+    return qs.toString(this.getValues());
```

Comment:
I think it's better to use `param.toUrlParams()` instead of mapping `name => value` directly - it provides some layer of abstraction for a case when param will need some different serialization logic in future.

Reply:
What I'm trying to achieve with this refactoring is moving out the serialization logic from Parameter/Parameters and into the query string module. The approach here is that a query string is itself the serialization "view", thus the logic resides there and the model simply reveals its data.